docs/.dagger/main.go

```diff
@@ -88,114 +77,88 @@ func (d Docs) Server() *dagger.Container {
-			WithWorkdir("/src").
-			WithExec([]string{"/changie", "merge"}).
-			Directory("/src")
-		return dag.Dirdiff().AssertEqual(ctx, before, after, []string{"CHANGELOG.md"})
-	})
-
-	eg.Go(func() (rerr error) {
-		ctx, span := Tracer().Start(ctx, "check that site builds")
-		defer func() {
-			if rerr != nil {
-				span.SetStatus(codes.Error, rerr.Error())
-			}
-			span.End()
-		}()
-		_, err := d.Site().Sync(ctx)
-		return err
-	})
-
-	// Go is already linted by engine:lint
-	// Python is already linted by sdk:python:lint
-	// TypeScript is already linted at sdk:typescript:lint
-	return eg.Wait()
+func (d Docs) CheckLint(ctx context.Context) error {
+	_, err := dag.Container().
+		From("tmknom/markdownlint:"+markdownlintVersion).
+		WithMountedDirectory("/src", d.Source).
+		WithMountedFile("/src/.markdownlint.yaml", d.Source.File(".markdownlint.yaml")).
+		WithMountedFile("/src/.markdownlintignore", d.Source.File("docs/.markdownlintignore")).
+		WithWorkdir("/src").
+		WithExec([]string{
+			"markdownlint",
+			"-c",
+			".markdownlint.yaml",
+			"--",
+			"./docs",
+			"README.md",
+		}).
+		Sync(ctx)
+	return err
 }
 
 // Regenerate the API schema and CLI reference docs
-func (d Docs) Generate(ctx context.Context) (*dagger.Directory, error) {
-	dir := dag.Directory().
-		WithDirectory("", d.GenerateSchema("")).
-		WithDirectory("", d.GenerateSchemaReference("")).
-		WithDirectory("", d.GenerateCli()).
-		WithDirectory("", d.GenerateConfigSchemas())
-
-	return dir, nil
-}
-
-// Regenerate the CLI reference docs
-func (d Docs) GenerateCli() *dagger.Directory {
-	generated := dag.DaggerCli().Reference(dagger.DaggerCliReferenceOpts{
-		Frontmatter:         cliZenFrontmatter,
-		IncludeExperimental: true,
-	})
-	return dag.Directory().WithFile(generatedCliZenPath, generated)
+func (d Docs) Generate(
+	ctx context.Context,
+	// Check that no generation was needed.
+	// Use this in CI, when it's too late to actually generate, but you need
+	// to enforce that developers remembered to do it.
+	// +optional
+	check bool,
+	// Dagger version to generate API docs for
+	// +optional
+	version string,
+) (*dagger.Changeset, error) {
+	src := d.Source
+	// 1. Generate the GraphQL schema
+	withGqlSchema := dag.Go(src).Env().
+		WithExec([]string{"go", "run", "./cmd/introspect", "--version=" + version, "schema"}, dagger.ContainerWithExecOpts{
+			RedirectStdout: "docs/docs-graphql/schema.graphqls",
+		}).
+		Directory(".")
+
+	// 2. Generate the API reference docs
+	//withApiReference := dag.Container().
+	//	From("node:22").
+	//	WithMountedDirectory("/src", withGqlSchema).
+	//	WithMountedDirectory("/mnt/spectaql", spectaql()).
+	//	WithWorkdir("/src/docs").
+	//	WithExec([]string{"yarn", "add", "file:/mnt/spectaql"}).
+	//	// -t specifies the target directory where spectaql will write the generated output
+	//	WithExec([]string{"yarn", "run", "spectaql", "./docs-graphql/config.yml", "-t", "./static/api/reference/"}).
+	//	Directory("/src")
```

Comment:
Are we intentionally not doing this anymore?

Reply:
Not intentional... 😬 Fixing